src/BSeries.jl

```diff
@@ -1453,4 +1456,419 @@ include("latexify.jl")
+This function generates all the equivalent trees 
+for a given level_sequence
+"""
+function equivalent_trees(array)
+    tree = rootedtree(array)
+    l = length(array)
+    #we get the permutations of the array
+    superarray = get_permutations(array)
+    lperm = length(superarray)
+    #eliminate all the arrays except for those for which the first element is equal to 1
+    for i in reverse(1:lperm) 
+        if superarray[i][1] != 1
+            splice!(superarray, i)
+        end
+    end
+    lperm = length(superarray)
+    #now, check if the levelsequence is the level sequence represents a RootedTree:
+    #the condition is that each number m cannot be followed by a number n such that
+    #n-m > 1. If that's the case, eliminate this array with 'splice!'
+    RTree_array = true
+    for i in reverse(1:lperm)
+        for j in 1:(l-1)
+            if (superarray[i][j+1] - superarray[i][j]) > 1
+                RTree_array = false
+            end
+        end
+        if RTree_array == false
+            splice!(superarray, i)
+        end
+        RTree_array = true
+    end
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        onetree = rootedtree(superarray[i])
+        isthesame_tree = onetree == tree
+        if isthesame_tree == false
+            splice!(superarray, i)
+        end
+    end
+    repeated_tree = false
+    lperm = length(superarray)
+    #we delete any repeated tree
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                repeated_tree = true
+            end
+        end 
+        if repeated_tree == true
+            splice!(superarray, i)
+        end
+        repeated_tree = false
+    end
+
+    return superarray
+end
+
+function get_permutations(arr)
+    perms = collect(permutations(arr))
+    return perms
+end
+
+function num_leafs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
+    l = length(arrays)
+    for i in 1:l
+        arrays[i] = vcat([1], arrays[i])
+    end
+    return arrays
+end
+
+#this function multplies the coefficient for its symmetry factor
+function renormalize_bseries(coefficient_array,thetrees)
+    l = length(coefficient_array)
+    for i in 1:l
+        factor = 0
+        #because of the librarys we are using, some packages are repeated
+        #Then, we specify that the symmetry function comes from RootedTrees
+        factor = symmetry(thetrees[i])
+        coefficient_array[i] = coefficient_array[i]*(1//factor)
+    end
+    return coefficient_array
+end
+
+"""
+This function tells up to what order a method is Energy Preserving.
+"""
+function energy_preserving_order(A,b)
+    rka = RungeKuttaMethod(A, b)
+    p = 0
+    not_energy_preserving = false
+    while not_energy_preserving == false 
+#generate bseries 
+        if is_energy_preserving(rka,p+1) == false
+            not_energy_preserving = true
+        end
+        p = p + 1
+    end
+    return p-1
+end
```

Comment:
This function (at least the version that accepts a BSeries as input) does need some way to stop in case you give it an exactly energy-preserving method like AVF.  Perhaps there could be a default maximum order, and if it reaches that it prints a message saying the method is EP to *at least* that order.  And the user could override the default if they want to check to higher order.

Reply:
I added this parameter